remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/ProtobufMetadataManagerImpl.java

```diff
@@ -95,16 +95,24 @@ void start() {
             getProtobufMetadataCacheConfig(globalConfiguration).build(),
             EnumSet.of(InternalCacheRegistry.Flag.USER, InternalCacheRegistry.Flag.PROTECTED, InternalCacheRegistry.Flag.PERSISTENT));
 
-      processSerializationContextInitializer(ServiceFinder.load(SerializationContextInitializer.class, globalConfiguration.classLoader()));
-      processSerializationContextInitializer(globalConfiguration.serialization().contextInitializers());
+      Collection<SerializationContextInitializer> initializers = globalConfiguration.serialization().contextInitializers();
+      if (initializers == null || initializers.isEmpty()) {
```

Comment:
I'm still not convinced this is necessary. What's the real benefit?

Consider this scenario:
- `x` SCI's auto generated and services created as per the new protostream default behaviour
- A single "manual" SCI implementation is created by the user

In this scenario, with your changes, it's now necessary for the user to manually configure all `x` SCIs in addition to the manual one.

Reply:
That is the intent of this PR, to force the user to either configure all SCIs manually (which is the old backward compat way) or go fully automatic. Manually SCI's written SCIs are normally found only in tests or our internal stuff, but the user should never need to do that.

I propose not mix the two approaches and if manual configuration of SCIs exists then it should be considered the single source of truth. It will basically be the way to turn off automatic CP scanning for SCIs which is not always the desirable way to do things. The CP might contain more than intended, even conflicting SCIs.  That's the typical case for tests, btw.

Reply:
> Manually written SCIs are normally found only in tests or our internal stuff, but the user should never need to do that.

Unfortunately manual SCIs are still the only way to configure manual marshallers for classes outside of the users control, e.g. UUID. This will remain the case until we have a solution for [IPROTO-137](https://issues.redhat.com/browse/IPROTO-137).

Reply:
@ryanemerson with this change we can continue to provide manually written SCIs, in both tests and production code, but we'll have to provide a service file for them if we don't plan to add them manually. So what is different ? :)

Reply:
Ok that's a fair point, I didn't really think about the user defining a service for the manual implementations. The only thing missing now, is that this behaviour needs to be documented :slightly_smiling_face: